### PR TITLE
fix: add 0 check for toRepay and toWithdraw compound

### LINKED ISSUE #1630: Compound 0 toRepay check

### PR DESCRIPTION
**Pull request by pakim249CAL**
# Pull Request

## Issue(s) fixed

This pull request fixes #1630, #1632

### CODE REVIEW

File: src/compound/PositionsManager.sol [hunk 1]
```diff
@@ -468,10 +468,10 @@ contract PositionsManager is IPositionsManager, MatchingEngine {
         if (_amount == 0) revert AmountIsZero();
         if (!marketStatus[_poolToken].isCreated) revert MarketNotCreated();
         if (marketPauseStatus[_poolToken].isRepayPaused) revert RepayIsPaused();
-        if (!userMembership[_poolToken][_onBehalf]) revert UserNotMemberOfMarket();
+        uint256 toRepay = Math.min(_getUserBorrowBalanceInOf(_poolToken, _onBehalf), _amount);
+        if (!userMembership[_poolToken][_onBehalf] || toRepay == 0) revert UserNotMemberOfMarket();
 
         _updateP2PIndexes(_poolToken);
```
Comment (MathisGD) [inline src/compound/PositionsManager.sol:474]: don't you need the indexes to be updated before using `_getUserBorrowBalanceInOf` ?

**Review by MathisGD — COMMENTED** [review]
I understand that the following situation can occur: a user is member of a market, but cannot repay more than 0 (due to roundings ?). Maybe we can have a quick test checking that now it reverts as expected.

**Review by MathisGD — COMMENTED** [review]
Can't the exact same thing happen in withdraw ?

**Comment by pakim249CAL** [comment]
> Can't the exact same thing happen in withdraw ?

Added an issue for this. #1632

File: src/compound/PositionsManager.sol [hunk 2]
```diff
@@ -471,7 +472,9 @@ contract PositionsManager is IPositionsManager, MatchingEngine {
         if (!userMembership[_poolToken][_onBehalf]) revert UserNotMemberOfMarket();
 
         _updateP2PIndexes(_poolToken);
+
```
Comment (MathisGD) [inline src/compound/PositionsManager.sol:475]:
```suggestion
```

**Review by MathisGD — APPROVED** [review]

File: src/compound/PositionsManager.sol [hunk 3]
```diff
@@ -446,6 +446,7 @@ contract PositionsManager is IPositionsManager, MatchingEngine {
 
         _updateP2PIndexes(_poolToken);
         uint256 toWithdraw = Math.min(_getUserSupplyBalanceInOf(_poolToken, _supplier), _amount);
+        if (toWithdraw == 0) revert UserNotMemberOfMarket();
```
Comment (MerlinEgalite) [inline src/compound/PositionsManager.sol:449]: I just realised that this line:
`if (!userMembership[_poolToken][_supplier]) revert UserNotMemberOfMarket();` is already doing the job and this why we did not implement it in Compound I think. 

Wdyt? Now I think it's really useless to add this check tbh

Comment (MathisGD) [reply src/compound/PositionsManager.sol:449]: I assumed that we saw the bug happening

>I understand that the following situation can occur: a user is member of a market, but cannot repay more than 0 (due to roundings ?). Maybe we can have a quick test checking that now it reverts as expected.

But if not (and we can prove it), then I agree that this is useless

Comment (pakim249CAL) [reply src/compound/PositionsManager.sol:449]: Shall I close the PR and issues then?

Comment (Rubilmax) [reply src/compound/PositionsManager.sol:449]: I was against introducing this at first so I can't be against closing this now 🤭
(ie I am in favor of closing it)

**Pull request closed without merging**